src/http.c

```diff
@@ -103,45 +104,51 @@ static int http_headers_complete_cb(http_parser *p) {
     um_http_req_t *req = p->data;
     req->state = headers_received;
     if (req->resp_cb != NULL) {
-        req->resp_cb(req, p->status_code, &req->resp_headers);
+        req->resp_cb(&req->resp, req->data);
     }
     return 0;
 }
 
 static int http_header_field_cb(http_parser *parser, const char *f, size_t len) {
     um_http_req_t *req = parser->data;
-    um_http_hdr *h = malloc(sizeof(um_http_hdr));
-    h->name = strndup(f, len);
-    LIST_INSERT_HEAD(&req->resp_headers, h, _next);
+    if (req->resp.headers == NULL) {
+        req->resp.headers = calloc(20, sizeof(um_http_hdr));
+    }
+    req->resp.headers[req->resp.nh].name = strndup(f, len);
     return 0;
 }
 
 static int http_header_value_cb(http_parser *parser, const char *v, size_t len) {
     um_http_req_t *req = parser->data;
-    um_http_hdr *h = LIST_FIRST(&req->resp_headers);
-    h->value = strndup(v, len);
+    req->resp.headers[req->resp.nh++].value = strndup(v, len);
+
     return 0;
 }
 
 static int http_status_cb(http_parser *parser, const char *status, size_t len) {
-    UM_LOG(VERB, "status = %d %*.*s", parser->status_code, (int) len, (int) len, status);
+    UM_LOG(VERB, "status = %d %.*s", parser->status_code, (int) len, status);
+    um_http_req_t *r = parser->data;
+    r->resp.code = (int) parser->status_code;
+    snprintf(r->resp.http_version, sizeof(r->resp.http_version), "%d.%d", parser->http_major, parser->http_minor);
```

Comment:
fwiw, HttpVersion on response is passed around as a String in Apple's framework:
`HTTPURLResponse.init?(url: URL, statusCode: Int, httpVersion HTTPVersion: String?, headerFields: [String : String]?)`